extensions/rich_text_components/Video/Video.js

```diff
@@ -25,19 +25,30 @@ oppia.directive('oppiaNoninteractiveVideo', [
       restrict: 'E',
       scope: {},
       templateUrl: 'richTextComponent/Video',
-      controller: ['$scope', '$attrs', function($scope, $attrs) {
-        var start = oppiaHtmlEscaper.escapedJsonToObj($attrs.startWithValue);
-        var end = oppiaHtmlEscaper.escapedJsonToObj($attrs.endWithValue);
+      controller: [
+        '$scope', '$attrs', 'explorationContextService', 'PAGE_CONTEXT',
+        'EDITOR_TAB_CONTEXT',
+        function(
+          $scope, $attrs, explorationContextService,
+          PAGE_CONTEXT, EDITOR_TAB_CONTEXT) {
+          var start = oppiaHtmlEscaper.escapedJsonToObj($attrs.startWithValue);
+          var end = oppiaHtmlEscaper.escapedJsonToObj($attrs.endWithValue);
 
-        $scope.autoplaySuffix = (oppiaHtmlEscaper.escapedJsonToObj(
-          $attrs.autoplayWithValue) ? '&autoplay=1' : '&autoplay=0');
-        $scope.videoId = oppiaHtmlEscaper.escapedJsonToObj(
-          $attrs.videoIdWithValue);
-        $scope.timingParams = '&start=' + start + '&end=' + end;
-        $scope.videoUrl = $sce.trustAsResourceUrl(
-          'https://www.youtube.com/embed/' + $scope.videoId + '?rel=0' +
-          $scope.timingParams + $scope.autoplaySuffix);
-      }]
+          $scope.autoplaySuffix = (oppiaHtmlEscaper.escapedJsonToObj(
+            $attrs.autoplayWithValue) ? '&autoplay=1' : '&autoplay=0');
+          $scope.videoId = oppiaHtmlEscaper.escapedJsonToObj(
+            $attrs.videoIdWithValue);
+          $scope.timingParams = '&start=' + start + '&end=' + end;
+          $scope.videoUrl = $sce.trustAsResourceUrl(
+            'https://www.youtube.com/embed/' + $scope.videoId + '?rel=0' +
+            $scope.timingParams + $scope.autoplaySuffix);
+          if (explorationContextService.getPageContext() ===
```

Comment:
This check comes up quite a bit. Maybe add a single method for it in explorationContextService?